lib/spack/spack/test/build_environment.py

```diff
@@ -0,0 +1,85 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os
+import unittest
+
+import spack
+from llnl.util.filesystem import join_path
+from spack.build_environment import dso_suffix, _static_to_shared_library
+from spack.util.executable import Executable
+
+
+class BuildEnvironmentTest(unittest.TestCase):
+
+    def setUp(self):
```

Comment:
can you use a `pytest` fixture for this?  We still have some old `unittest` style tests but we're trying to switch to `pytest`.  You can probably just make a `build_environment` fixture that does all the same stuff as this, and yields a named tuple with `cc`, `ld`, etc.  It should probably also unset the environment variables after yielding so as not to interfere with other tests.